app/models/metric/ci_mixin/capture.rb

```diff
@@ -17,6 +17,20 @@ def queue_name_for_metrics_collection
     ems.metrics_collector_queue_name
   end
 
+  def split_capture_intervals(start_time, end_time)
+    intervals = []
+    start_hour = start_time
+    while start_hour != end_time
+      #TODO: replace 1.day with a configurable threshold defined by provider type
+      end_hour = start_hour + 1.day
+      end_hour = end_time if end_hour > end_time
+      intervals << [start_hour, end_hour]
+      start_hour = end_hour
+    end
```

Comment:
This feels very manual.  I think you can do this with.

```ruby
(start_time..end_time).step_value(1.day)
```

If you definitely need them in start/end pairs, just tack on a `each_cons(2)`

```ruby
(start_time..end_time).step_value(1.day).each_cons(2)
```